packages/arb-token-bridge-ui/src/hooks/useCCTP.ts

```diff
@@ -0,0 +1,324 @@
+    mode: 'recklesslyUnprepared',
+    chainId: targetChainId
+  })
+
+  const deposit = useCallback(
+    async (amount: BigNumber) => {
+      const depositTx = await depositForBurn({
+        recklesslySetUnpreparedArgs: [
+          amount,
+          targetChainDomain,
+          destinationAddress as `0x${string}`,
+          USDCContractAddress
+        ]
+      })
+
+      const txReceipt = await depositTx.wait()
+      const eventTopic = utils.keccak256(
+        utils.toUtf8Bytes('MessageSent(bytes)')
+      )
+      const log = txReceipt.logs.find(l => l.topics[0] === eventTopic)
+
+      if (!log) return
+
+      const messageBytes = utils.defaultAbiCoder.decode(['bytes'], log.data)[0]
+      const attestationHash = utils.keccak256(messageBytes)
+
+      return {
+        attestationHash,
+        messageBytes
+      } as {
+        attestationHash: `0x${string}`
+        messageBytes: `0x${string}`
+      }
+    },
+    [
+      targetChainDomain,
+      destinationAddress,
+      USDCContractAddress,
+      receiveMessage,
+      depositForBurn
+    ]
+  )
+
+  const fetchAttestation = useCallback(
+    async (attestationHash: `0x${string}`) => {
+      let attestationResponse: AttestationResponse = {
+        status: 'pending_confirmations',
+        attestation: null
+      }
+
+      while (true) {
+        const response = await fetch(
+          `${attestation_API_URL}/attestations/${attestationHash}`
+        )
+
+        attestationResponse = await response.json()
+        if (attestationResponse.status === 'complete') {
+          break
+        }
+
+        await new Promise(r => setTimeout(r, 5000))
+      }
+
+      return attestationResponse.attestation
+    },
+    []
+  )
+
+  const redeem = useCallback(
+    async ({
+      messageBytes,
+      attestation
+    }: {
+      messageBytes: `0x${string}`
+      attestation: `0x${string}`
+    }) => {
+      const receiveTx = await receiveMessage({
+        recklesslySetUnpreparedArgs: [messageBytes, attestation]
+      })
+      return await receiveTx.wait()
+    },
+    [receiveMessage]
+  )
+
+  const updateUSDCBalances = useCallback((address: `0x${string}` | string) => {
+    const l1Address = getL1AddressFromAddress(address)
+
+    updateErc20L1Balance([l1Address.toLocaleLowerCase()])
+
+    if (isTokenMainnetUSDC(l1Address)) {
+      updateErc20L2Balance([
+        CommonAddress.ArbitrumOne.USDC,
+        CommonAddress.ArbitrumOne['USDC.e']
+      ])
+    } else if (isTokenGoerliUSDC(l1Address)) {
+      updateErc20L2Balance([
+        CommonAddress.ArbitrumGoerli.USDC,
+        CommonAddress.ArbitrumGoerli['USDC.e']
+      ])
+    }
+  }, [])
+
+  const approveUSDC = useCallback(
+    async (signer: Signer, amount: BigNumber) => {
+      const contract = ERC20__factory.connect(USDCContractAddress, signer)
+      const tx = await contract.functions.approve(
+        tokenMessengerContractAddress,
+        amount
+      )
+      await tx.wait()
```

Comment:
Hm, maybe we just wanna do 

```suggestion
      return contract.functions.approve(
        tokenMessengerContractAddress,
        amount
      )
```

and then be able to have the tx success/failure handled in the call site

Reply:
Done